source/com.microsoft.tfs.core/rest_core/com/microsoft/alm/client/TeeClientHandler.java

```diff
@@ -0,0 +1,371 @@
+        final String acceptType =
+            getMediaTypeWithQualityHeaderValue(acceptMediaType, negotiateRequestVersion(locationId, version));
+
+        final HttpMethodBase request;
+        if (shouldOverrideHttpMethod(method)) {
+            request = createHttpMethod(HttpMethod.POST, target);
+            request.setRequestHeader(VssHttpHeaders.HTTP_METHOD_OVERRIDE, method.toString());
+        } else {
+            request = createHttpMethod(method, target);
+        }
+
+        request.setRequestHeader(VssHttpHeaders.ACCEPT, acceptType);
+        request.setRequestHeader(VssHttpHeaders.TFS_VERSION, CoreVersionInfo.getVersion());
+
+        if (request instanceof EntityEnclosingMethod) {
+            final String contentType = getMediaTypeWithQualityHeaderValue(contentMediaType);
+            final EntityEnclosingMethod entityMethod = (EntityEnclosingMethod) request;
+
+            RequestEntity entity = null;
+
+            if (value != null && (value instanceof InputStream)) {
+                entity = new InputStreamRequestEntity((InputStream) value, contentMediaType);
+            } else {
+                final String content;
+                if (value != null) {
+                    content = JsonHelper.serializeRequestToString(value);
+                } else {
+                    content = StringUtil.EMPTY;
+                }
+
+                try {
+                    entity = new StringRequestEntity(content, contentType, StringUtil.UTF8_CHARSET);
+                } catch (final UnsupportedEncodingException e) {
+                    // Note that this exception cannot happen at this point
+                    log.error("Illegal content encoding found.", e); //$NON-NLS-1$
+                }
+            }
+
+            entityMethod.setRequestEntity(entity);
+            entityMethod.setRequestHeader(VssHttpHeaders.CONTENT_LENGTH, String.valueOf(entity.getContentLength()));
+        }
+
+        return new TeeRestRequest(request);
+    }
+
+    private URI createTarget(
+        final UUID locationId,
+        final Map<String, Object> routeValues,
+        final Map<String, String> queryParameters) {
+
+        final ApiResourceLocation location = getLocation(locationId);
+        if (location == null) {
+            throw new VssResourceNotFoundException(locationId, getBaseUrl(), getLastException());
+        }
+
+        final Map<String, Object> dictionary =
+            toRouteDictionary(routeValues, location.getArea(), location.getResourceName());
+
+        final String routeTemplate = location.getRouteTemplate();
+        final String actualTemplate = replaceRouteValues(routeTemplate, dictionary);
+        final URI target = URIUtils.resolve(getBaseUrl(), actualTemplate);
+
+        return URIUtils.addQueryParameters(target, queryParameters);
+    }
+
+    private String replaceRouteValues(final String template, final Map<String, Object> routeValues) {
```

Comment:
P3:  Don't we have a very similar method in the other RestHandler?  Maybe next release we could consolidate this and extract this method to the parent class?